### PR TITLE
Routes for Transfer / Burn

### PR DESCRIPTION
**Pull request by redpartyhat**

### CODE REVIEW

**Review by superzordon — APPROVED** [review]
LGTM 🚀

File: routes/nft.go [hunk 1]
```diff
@@ -1338,7 +1340,389 @@ func (fes *APIServer) _bidEntryToResponse(bidEntry *lib.NFTBidEntry, postEntryRe
...
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"TransferNFT: Error could not find the NFT an NFT with postHash %v and serialNumber %d",
+			requestData.NFTPostHashHex, requestData.SerialNumber))
+		return
+
+	} else if nftEntry.IsForSale {
+		_AddBadRequestError(ww, fmt.Sprintf("TransferNFT: Cannot transfer NFT that is for sale."))
+		return
+	}
+
```
Comment (lazynina) [inline routes/nft.go:1444]: Should we check that the sender owns the NFT somewhere in this function?

File: routes/nft.go [hunk 2]
```diff
@@ -1338,7 +1340,389 @@ func (fes *APIServer) _bidEntryToResponse(bidEntry *lib.NFTBidEntry, postEntryRe
...
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"AcceptNFTTransfer: Error could not find the NFT an NFT with postHash %v and serialNumber %d",
+			requestData.NFTPostHashHex, requestData.SerialNumber))
+		return
+
+	} else if !nftEntry.IsPending {
+		_AddBadRequestError(ww, fmt.Sprintf("AcceptNFTTransfer: This NFT is not a pending transfer."))
+		return
+	}
+
```
Comment (lazynina) [inline routes/nft.go:1574]: Can we validate that updaterPublicKeyBase58Check is the recipient of the NFT transfer for this serial number?

File: routes/nft.go [hunk 3]
```diff
@@ -1338,7 +1340,389 @@ func (fes *APIServer) _bidEntryToResponse(bidEntry *lib.NFTBidEntry, postEntryRe
...
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"BurnNFT: Error could not find the NFT an NFT with postHash %v and serialNumber %d",
+			requestData.NFTPostHashHex, requestData.SerialNumber))
+		return
+
+	} else if nftEntry.IsForSale {
+		_AddBadRequestError(ww, fmt.Sprintf("BurnNFT: Cannot burn an NFT that is for sale."))
+		return
+	}
+
```
Comment (lazynina) [inline routes/nft.go:1693]: Should we validate that the updater is currently the owner of this NFT?

**Review by lazynina — COMMENTED** [review]
Some comments about additional validations, but other than that, this looks good to go.

File: routes/nft.go (hunk 1 above)
Comment (redpartyhat) [reply routes/nft.go:1444]: Added and tested :+1:

File: routes/nft.go (hunk 2 above)
Comment (redpartyhat) [reply routes/nft.go:1581]: Added and tested :+1:

File: routes/nft.go (hunk 3 above)
Comment (redpartyhat) [reply routes/nft.go:1707]: Added and tested :+1:

**Pull request merged**